blockcerts/types/types.go

```diff
@@ -0,0 +1,21 @@
+package types
+
+import "github.com/spacemeshos/go-spacemesh/common/types"
+
+type BlockCertificate struct {
+    BlockID               types.BlockID
+    LayerID               types.LayerID
+    TerminationSignatures []BlockSignature
+}
+type BlockSignature struct {
+    SignerNodeID         types.NodeID
+    SignerRoleProof      []byte
+    SignerCommitteeSeats uint16
+
+    BlockIDSignature []byte
+}
+type BlockSignatureMsg struct {
+    LayerID types.LayerID
+    BlockID types.BlockID
+    BlockSignature
+}
```

Comment:
I don't understand the question. It's not in both of them.
We need the LayerID and BlockID the message structure to provide completeness such that we can validate the message upon receiving it without making assumptions about what layer/block it's referring to.
We don't need it in the BlockSignature struct because the BlockSignature struct will always be stored in a context where the BlockID that was signed is obvious. (e.g. in a BlockSignatureMsg or in a signature tracker which is counting the number of signatures for a specific block/layer id)

Reply:
i mistook BlockSignature as BlockCertificate. sorry about the confusion.

but i don't get in general what does LayerID has to do with the certificate? the certificate is about a block. why do we care about the layer ID?

Reply:
ok. it is needed in the Validate() call to verify proof. 

but i actually don't think we should blindly trust the layer ID specified in the msg. we have BlockID->LayerID stored in db and should look it up from there to avoid malicious message.